includes/server/class-llms-rest-access-plans-controller.php

```diff
@@ -0,0 +1,1081 @@
+	 * Retrieves an array of arguments for the delete endpoint
+	 *
+	 * @since [version]
+	 *
+	 * @return array Delete endpoint arguments.
+	 */
+	public function get_delete_item_args() {
+		return array();
+	}
+
+	/**
+	 * Whether the delete should be forced
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool True if the delete should be forced, false otherwise.
+	 */
+	protected function is_delete_forced( $request ) {
+		return true;
+	}
+
+	/**
+	 * Whether the trash is supported
+	 *
+	 * @since [version]
+	 *
+	 * @return bool True if the trash is supported, false otherwise.
+	 */
+	protected function is_trash_supported() {
+		return false;
+	}
+
+	/**
+	 * Check if a given request has access to create an item
+	 *
+	 * @since [version]
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function create_item_permissions_check( $request ) {
+
+		$can_create = parent::create_item_permissions_check( $request );
+
+		// If current user cannot create the item because of authorization, check if the current user can edit the "parent" course/membership.
+		if ( llms_rest_is_authorization_required_error( $can_create ) ) {
+
+			$post_type_object = get_post_type_object( get_post_type( $request['post_id'] ) );
+
+			if ( ! current_user_can( $post_type_object->cap->edit_post, $request['post_id'] ) ) {
+				return llms_rest_authorization_required_error(
+					sprintf(
+						// Translators: %s = The post type name.
+						__( 'Sorry, you are not allowed to create %s as this user.', 'lifterlms' ),
+						$post_type_object->labels->name
+					)
+				);
+			}
+
+			$can_create = true;
+		}
+
+		return $can_create;
+	}
+
+	/**
+	 * Check if a given request has access to update an item
+	 *
+	 * @since [version]
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function update_item_permissions_check( $request ) {
+
+		$can_update = parent::update_item_permissions_check( $request );
+
+		// If current user cannot edit the item because of authorization, check if the current user can edit the "parent" course/membership.
+		if ( llms_rest_is_authorization_required_error( $can_update ) ) {
+
+			$access_plan = $this->get_object( (int) $request['id'] );
+
+			if ( is_wp_error( $access_plan ) ) {
+				return $can_update;
+			}
+
+			$product_id               = $access_plan->get( 'product_id' );
+			$product_post_type_object = get_post_type_object( get_post_type( $product_id ) );
+
+			if ( ! current_user_can( $product_post_type_object->cap->edit_post, $product_id ) ) {
+				return llms_rest_authorization_required_error(
+					sprintf(
+						// Translators: %s = The post type name.
+						__( 'Sorry, you are not allowed to update %s as this user.', 'lifterlms' ),
+						$access_plan->get_post_type_label( 'name' )
+					)
+				);
+			}
+
+			$can_update = true;
+		}
```

Comment:
This particular bit is repeated for the other permission checks, I think it would be good to move this check into a protected method so we can reuse it instead of repeating it.

Reply:
Right, refactored a little bit.